HeifImagePlugin.py

```diff
@@ -174,10 +177,14 @@ def _save(im, fp, filename):
         # disbled due to errors in libheif encoder
         raise IOError("cannot write mode P as HEIF")
 
+    if im.mode == '1':
```

Comment:
Could we define enum or at least constants for that?
```python
class ImageMode(str, Enum):
    binary = '1'
    grayscale = 'L'
 
...

if im.mode == ImageMode.binary:
    im = im.convert(ImageMode.grayscale)
```

Reply:
It's something related to Pillow API itself (which is quite ancient), since it should be used not in `heif-image-plugin` itself.